src/settingsdialog.cpp

```diff
@@ -58,10 +67,55 @@ SettingsDialog::SettingsDialog(QWidget* parent)
     };
     auto lastExtraLibsWidget = setupMultiPath(ui->extraLibraryPaths, ui->extraLibraryPathsLabel, ui->lineEditApplicationPath);
     setupMultiPath(ui->debugPaths, ui->debugPathsLabel, lastExtraLibsWidget);
+
+    const auto configGroups = config().groupList();
+    auto configfile = config();
+    for (const auto& configName : configGroups) {
+        if (configfile.hasGroup(configName)) {
```

Comment:
this looks like it's duplicating a lot of code with the record page, can we share the logic somehow?

Reply:
I created a custom widget that does this